wsagent/che-wsagent-core/src/main/java/org/eclipse/che/wsagent/server/CheWebSocketEndpoint.java

```diff
@@ -39,6 +58,67 @@ public CheWebSocketEndpoint(
 
   @Override
   protected String getEndpointId() {
-    return "ws-agent-websocket-endpoint";
+    return ENDPOINT_ID;
+  }
+
+  public static class CheWebSocketEndpointConfiguration
+      implements RequestProcessorConfigurator.Configuration {
+
+    private final ExecutorService executor;
+
+    @Inject
+    public CheWebSocketEndpointConfiguration(
+        @Named("che.core.jsonrpc.major_executor") ExecutorService executor) {
+      this.executor = executor;
+    }
+
+    @Override
+    public String getEndpointId() {
+      return ENDPOINT_ID;
+    }
+
+    @Override
+    public ExecutorService getExecutorService() {
+      return executor;
+    }
+  }
+
+  @Singleton
+  public static class CheWebSocketEndpointExecutorServiceProvider
+      implements Provider<ExecutorService> {
+
+    private final ThreadPoolExecutor executor;
+
+    @Inject
+    public CheWebSocketEndpointExecutorServiceProvider(
+        @Named("che.core.jsonrpc.processor_core_pool_size") int corePoolSize,
+        @Named("che.core.jsonrpc.processor_max_pool_size") int maxPoolSize,
+        @Named("che.core.jsonrpc.processor_queue_capacity") int queueCapacity) {
+      ThreadFactory factory =
```

Comment:
Is there a place common to agent and master where we could put this? Both the `factory` and `executor` are instantiated in the very same way here and in `CheMajorWebSocketEndpointExecutorServiceProvider` that is non-trivial enough for it to be prone to oversight when changed in the future...